src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -2,24 +2,45 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 public class ReadWriteList<E> {
-    private List<E> list = new ArrayList<>();
-    private ReadWriteLock lock = new ReentrantReadWriteLock();
+    private final List<E> list = new ArrayList<>();
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        Lock writeLock = lock.writeLock();
+        writeLock.lock();
+        try {
+            list.add(element);
+        } finally {
+            writeLock.unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        E element;
+        Lock readLock = lock.readLock();
+        readLock.lock();
+        try {
+            element = list.get(index);
```

Comment:
to make your code more concise
```suggestion
            return list.get(index);
```